yb-voyager/src/migassessment/gather.go

```diff
@@ -0,0 +1,581 @@
+}
+
+func findGatherMetadataScriptPath(dbType string) (string, error) {
+	var defaultScriptPath string
+	switch dbType {
+	case POSTGRESQL:
+		defaultScriptPath = "/etc/yb-voyager/gather-assessment-metadata/postgresql/yb-voyager-pg-gather-assessment-metadata.sh"
+	case ORACLE:
+		defaultScriptPath = "/etc/yb-voyager/gather-assessment-metadata/oracle/yb-voyager-oracle-gather-assessment-metadata.sh"
+	default:
+		panic(fmt.Sprintf("invalid source db type %q", dbType))
+	}
+
+	homebrewVoyagerDir := fmt.Sprintf("yb-voyager@%s", utils.YB_VOYAGER_VERSION)
+	possiblePathsForScript := []string{
+		defaultScriptPath,
+		filepath.Join("/", "opt", "homebrew", "Cellar", homebrewVoyagerDir, utils.YB_VOYAGER_VERSION, defaultScriptPath),
+		filepath.Join("/", "usr", "local", "Cellar", homebrewVoyagerDir, utils.YB_VOYAGER_VERSION, defaultScriptPath),
+	}
+
+	for _, path := range possiblePathsForScript {
+		if utils.FileOrFolderExists(path) {
+			log.Infof("found the gather assessment metadata script at: %s", path)
+			return path, nil
+		}
+	}
+
+	return "", fmt.Errorf("script not found in possible paths: %v", possiblePathsForScript)
+}
+
+func runGatherAssessmentMetadataScript(scriptPath string, envVars []string, workingDir string, scriptArgs ...string) error {
+	cmd := exec.Command(scriptPath, scriptArgs...)
+	log.Infof("running script: %s", cmd.String())
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, envVars...)
+	cmd.Dir = workingDir
+	cmd.Stdin = os.Stdin
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("error creating stdout pipe: %w", err)
+	}
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("error creating stderr pipe: %w", err)
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("error starting gather assessment metadata script: %w", err)
+	}
+
+	go func() {
+		scanner := bufio.NewScanner(stderr)
+		for scanner.Scan() {
+			log.Errorf("[stderr of script]: %s", scanner.Text())
+			fmt.Printf("%s\n", scanner.Text())
+		}
+	}()
+
+	scanner := bufio.NewScanner(stdout)
+	for scanner.Scan() {
+		log.Infof("[stdout of script]: %s", scanner.Text())
+		fmt.Printf("%s\n", scanner.Text())
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				if status.ExitStatus() == 2 {
+					log.Infof("Exit without error as user opted not to continue in the script.")
+					os.Exit(0)
+				}
+			}
+		}
+		return fmt.Errorf("error waiting for gather assessment metadata script to complete: %w", err)
+	}
+	return nil
+}
+
+// runGatherAssessmentMetadataScriptBuffered runs the metadata collection script with buffered output
+// and sends progress updates to the provided channel. This version is safe for parallel execution.
+func runGatherAssessmentMetadataScriptBuffered(
+	scriptPath string,
+	envVars []string,
+	nodeName string,
+	displayName string,
```

Comment:
displayName is only the problem of displaying, not of gathering, so it seems to me that storing the display name should be the responsibility of the progressTracker alone, and only the nodeName (or the nodeId essentially) should be passed around everywhere. 


I don't have clarity on this, but let's discuss

Reply:
Makes sense. Refactored the code around it. Only progressTracker now creates and saves displayName.